src/editors/containers/ProblemEditor/data/OLXParser.js

```diff
@@ -0,0 +1,382 @@
+import { XMLParser, XMLBuilder } from 'fast-xml-parser';
+import _ from 'lodash-es';
+import { ProblemTypeKeys } from '../../../data/constants/problem';
+import { StrictDict } from '../../../utils';
+
+export const indexToLetterMap = [...Array(26)].map((val, i) => String.fromCharCode(i + 65));
+
+export const OLXProblemTypes = StrictDict({
+  TEXTINPUT: 'stringresponse',
+  NUMERIC: 'numericalresponse',
+  DROPDOWN: 'optionresponse',
+  MULTISELECT: 'choiceresponse',
+  SINGLESELECT: 'multiplechoiceresponse',
+});
+
+export const nonQuestionKeys = [
+  'responseparam',
+  'formulaequationinput',
+  'correcthint',
+  '@_answer',
+  'optioninput',
+  'checkboxgroup',
+  'choicegroup',
+  'additional_answer',
+  'stringequalhint',
+  'textline',
+  '@_type',
+  'formulaequationinput',
+  'numericalresponse',
+  'demandhint',
+];
+
+export class OLXParser {
+  constructor(olxString) {
+    const options = {
+      ignoreAttributes: false,
+      alwaysCreateTextNode: true,
+      // preserveOrder: true
+    };
+    const parser = new XMLParser(options);
+    this.parsedOLX = parser.parse(olxString);
```

Comment:
Nit:
Perhaps we can extract the problem block here at a singular place instead of each of the methods.

```suggestion
    this.parsedOLX = parser.parse(olxString);
    let problem = {};
    if (_.has(this.parsedOLX, 'problem')) {
        const { problem } = this.parsedOLX;
    }
```


We can suitably modify the [check below](https://github.com/open-craft/frontend-lib-content-components/blob/c53a13780a6338702044fa03af9d009ba6d2b0c4/src/editors/containers/ProblemEditor/data/OLXParser.js#L335-L337) to 

```
if(_.isEmpty(this.problem)){
    return {};
}
```

Reply:
Done!